v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -160,10 +164,17 @@ export async function parseHardhatSpecialArguments(
   let showStackTraces: boolean = isCi();
   let help: boolean = false;
   let version: boolean = false;
+  let init: boolean = false;
 
   for (let i = 0; i < cliArguments.length; i++) {
     const arg = cliArguments[i];
 
+    if (arg === "init") {
```

Comment:
Let's switch to `--init` instead. `init` would be too ambiguous for this naive parsing logic.

Reply:
Done [here](https://github.com/NomicFoundation/hardhat/pull/5425/commits/3a04f738ef6c6e8dafd0f1141459824172b058d2)